test/Core.Test/AdminConsole/OrganizationAuth/UpdateOrganizationAuthRequestCommandTests.cs

```diff
@@ -0,0 +1,101 @@
+﻿using Bit.Core.AdminConsole.OrganizationAuth;
+using Bit.Core.Auth.Entities;
+using Bit.Core.Auth.Models.Api.Request.AuthRequest;
+using Bit.Core.Auth.Services;
+using Bit.Core.Entities;
+using Bit.Core.Enums;
+using Bit.Core.Repositories;
+using Bit.Core.Services;
+using Bit.Test.Common.AutoFixture;
+using Bit.Test.Common.AutoFixture.Attributes;
+using NSubstitute;
+using NSubstitute.ReturnsExtensions;
+using Xunit;
+
+namespace Bit.Core.Test.AdminConsole.OrganizationAuth;
+
+[SutProviderCustomize]
+public class UpdateOrganizationAuthRequestCommandTests
+{
+    [Theory]
+    [BitAutoData]
+    public async Task UpdateOrgAuthRequest_Approved_SendEmail_Success(
+        SutProvider<UpdateOrganizationAuthRequestCommand> sutProvider, Guid requestId, Guid userId,
+        bool requestApproved, string encryptedUserKey)
+    {
+        var responseDate = DateTime.UtcNow;
+        const string email = "email";
+        const DeviceType deviceType = DeviceType.ChromeBrowser;
+        const string deviceIdentifier = "Intel Mac OSX 10_15_7";
+        const string deviceTypeIdentifier = "Chrome - Intel Mac OSX 10_15_7";
+        const string requestIpAddress = "127.0.0.1";
+
+        sutProvider.GetDependency<IAuthRequestService>()
+            .UpdateAuthRequestAsync(requestId, userId,
+                Arg.Is<AuthRequestUpdateRequestModel>(x =>
+                    x.RequestApproved == requestApproved && x.Key == encryptedUserKey))
+            .Returns(new AuthRequest()
+            {
+                UserId = userId,
+                Approved = true,
+                ResponseDate = responseDate,
+                RequestDeviceType = deviceType,
+                RequestDeviceIdentifier = deviceIdentifier,
+                RequestIpAddress = requestIpAddress,
+            });
+
+        sutProvider.GetDependency<IUserRepository>()
+            .GetByIdAsync(userId)
+            .Returns(new User() { Email = email });
+
+        await sutProvider.Sut.UpdateAsync(requestId, userId, requestApproved, encryptedUserKey);
+
+        await sutProvider.GetDependency<IUserRepository>().Received(1).GetByIdAsync(userId);
+        await sutProvider.GetDependency<IMailService>().Received(1)
+            .SendTrustedDeviceAdminApprovalEmailAsync(email, responseDate, requestIpAddress, deviceTypeIdentifier);
+    }
```

Comment:
This looks great but I think it can be improved by having `BitAutoData` generate values for all the parameters used in the test.
Here's an example below on how to achieve that:

```suggestion
    public async Task UpdateOrgAuthRequest_Approved_SendEmail_Success(
        DateTime responseDate, string email, DeviceType deviceType, string deviceIdentifier,
        string requestIpAddress, Guid requestId, Guid userId, bool requestApproved,
        string encryptedUserKey, SutProvider<UpdateOrganizationAuthRequestCommand> sutProvider)
    {
        var expectedDeviceTypeIdentifier = $"{deviceType} - {deviceIdentifier}";

        sutProvider.GetDependency<IAuthRequestService>()
            .UpdateAuthRequestAsync(requestId, userId,
                Arg.Is<AuthRequestUpdateRequestModel>(x =>
                    x.RequestApproved == requestApproved && x.Key == encryptedUserKey))
            .Returns(new AuthRequest()
            {
                UserId = userId,
                Approved = true,
                ResponseDate = responseDate,
                RequestDeviceType = deviceType,
                RequestDeviceIdentifier = deviceIdentifier,
                RequestIpAddress = requestIpAddress,
            });

        sutProvider.GetDependency<IUserRepository>()
            .GetByIdAsync(userId)
            .Returns(new User() { Email = email });

        await sutProvider.Sut.UpdateAsync(requestId, userId, requestApproved, encryptedUserKey);

        await sutProvider.GetDependency<IUserRepository>().Received(1).GetByIdAsync(userId);
        await sutProvider.GetDependency<IMailService>().Received(1)
            .SendTrustedDeviceAdminApprovalEmailAsync(email, responseDate, requestIpAddress, expectedDeviceTypeIdentifier);
    }
```

I removed the manual initialization of the const strings and added those properties as parameters of the unit test.
I usually leave `SutProvider` as the last parameter so the other parameters can have a constant value be set using for example `[BitAutoData("email")]`

Reply:
Updated! Thanks again.